src/generic_solver.cpp

```diff
@@ -478,31 +485,81 @@ Sort GenericSolver::make_sort(SortKind sk, const SortVec & sorts) const
 
 Sort GenericSolver::make_sort(const DatatypeDecl & d) const
 {
-  throw NotImplementedException("Generic Solvers do not support datatypes");
+  std::string dtName = (*datatypedecl_name_map)[d];
```

Comment:
Before this line, please add an assertion that `d` exists in the map. It is very useful, because in case of a bug these assertions help understand what's going on. Also, if `d` happens to not exist in the map, this line will add it with some default value, and we don't want that.

ere is an example for such an assertion:
https://github.com/makaimann/smt-switch/pull/251/files#diff-08c7035034a6ef997827cf24e4181dd3e3de5e1ebc1ebe19c220dd92ec52624fR71